<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_51) on Fri Jun 28 19:01:13 EEST 2013 -->
<TITLE>
GeneticCode
</TITLE>

<META NAME="date" CONTENT="2013-06-28">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="GeneticCode";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/GeneticCode.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../dr/evolution/datatype/GeneralDataType.State.html" title="class in dr.evolution.datatype"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../dr/evolution/datatype/HiddenCodons.html" title="class in dr.evolution.datatype"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?dr/evolution/datatype/GeneticCode.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="GeneticCode.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
dr.evolution.datatype</FONT>
<BR>
Class GeneticCode</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>dr.evolution.datatype.GeneticCode</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD><A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public final class <B>GeneticCode</B><DT>extends java.lang.Object<DT>implements <A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></DL>
</PRE>

<P>
A set of standard genetic codes.
<P>

<P>
<DL>
<DT><B>Version:</B></DT>
  <DD>$Id: GeneticCode.java,v 1.11 2005/05/24 20:25:56 rambaut Exp $</DD>
<DT><B>Author:</B></DT>
  <DD>Andrew Rambaut, Alexei Drummond</DD>
</DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#ALT_YEAST">ALT_YEAST</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#ALT_YEAST_ID">ALT_YEAST_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#ASCIDIAN_MT">ASCIDIAN_MT</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#ASCIDIAN_MT_ID">ASCIDIAN_MT_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#BACTERIAL">BACTERIAL</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#BACTERIAL_ID">BACTERIAL_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#BLEPHARISMA_NUC">BLEPHARISMA_NUC</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#BLEPHARISMA_NUC_ID">BLEPHARISMA_NUC_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#CILIATE">CILIATE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#CILIATE_ID">CILIATE_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#codeTable">codeTable</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#ECHINODERM_MT">ECHINODERM_MT</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#ECHINODERM_MT_ID">ECHINODERM_MT_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#EUPLOTID_NUC">EUPLOTID_NUC</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#EUPLOTID_NUC_ID">EUPLOTID_NUC_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#FLATWORM_MT">FLATWORM_MT</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#FLATWORM_MT_ID">FLATWORM_MT_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#GENETIC_CODE">GENETIC_CODE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#GENETIC_CODE_DESCRIPTIONS">GENETIC_CODE_DESCRIPTIONS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Descriptions of the standard genetic code tables from GENBANK</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#GENETIC_CODE_NAMES">GENETIC_CODE_NAMES</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Names of the standard genetic code tables from GENBANK</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#GENETIC_CODE_TABLES">GENETIC_CODE_TABLES</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Standard genetic code tables from GENBANK
 Nucleotides go A, C, G, T - Note: this is not the order used by the Genbank web site
 With the first codon position most significant (i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A>[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#GENETIC_CODES">GENETIC_CODES</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#geneticCode">geneticCode</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#INVERTEBRATE_MT">INVERTEBRATE_MT</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#INVERTEBRATE_MT_ID">INVERTEBRATE_MT_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#MOLD_PROTOZOAN_MT">MOLD_PROTOZOAN_MT</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#MOLD_PROTOZOAN_MT_ID">MOLD_PROTOZOAN_MT_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#MYCOPLASMA">MYCOPLASMA</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#MYCOPLASMA_ID">MYCOPLASMA_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#NO_STOPS">NO_STOPS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#NO_STOPS_ID">NO_STOPS_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#UNIVERSAL">UNIVERSAL</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#UNIVERSAL_ID">UNIVERSAL_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constants used to refer to the built in code tables</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#VERTEBRATE_MT">VERTEBRATE_MT</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#VERTEBRATE_MT_ID">VERTEBRATE_MT_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#YEAST">YEAST</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#YEAST_ID">YEAST_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#GeneticCode(int)">GeneticCode</A></B>(int&nbsp;geneticCode)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getAmbiguousCodonFromAminoAcidState(int)">getAmbiguousCodonFromAminoAcidState</A></B>(int&nbsp;aminoAcid)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;char</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getAminoAcidChar(int)">getAminoAcidChar</A></B>(int&nbsp;codonState)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the char associated with AminoAcid represented by codonState.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getAminoAcidState(int)">getAminoAcidState</A></B>(int&nbsp;codonState)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the state associated with AminoAcid represented by codonState.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;char[][]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getCodonsFromAminoAcidChar(char)">getCodonsFromAminoAcidChar</A></B>(char&nbsp;aminoAcidChar)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;char[][]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getCodonsFromAminoAcidState(int)">getCodonsFromAminoAcidState</A></B>(int&nbsp;aminoAcidState)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getDescription()">getDescription</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the description of the genetic code</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getName()">getName</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the name of the genetic code</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getStopCodonCount()">getStopCodonCount</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the number of terminator amino acids.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#getStopCodonIndices()">getStopCodonIndices</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../dr/evolution/datatype/GeneticCode.html#isStopCodon(int)">isStopCodon</A></B>(int&nbsp;codonState)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Note that the state is the canonical state (generated combinatorially)</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="GENETIC_CODE"><!-- --></A><H3>
GENETIC_CODE</H3>
<PRE>
public static final java.lang.String <B>GENETIC_CODE</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.GENETIC_CODE">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="UNIVERSAL_ID"><!-- --></A><H3>
UNIVERSAL_ID</H3>
<PRE>
public static final int <B>UNIVERSAL_ID</B></PRE>
<DL>
<DD>Constants used to refer to the built in code tables
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.UNIVERSAL_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="VERTEBRATE_MT_ID"><!-- --></A><H3>
VERTEBRATE_MT_ID</H3>
<PRE>
public static final int <B>VERTEBRATE_MT_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.VERTEBRATE_MT_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="YEAST_ID"><!-- --></A><H3>
YEAST_ID</H3>
<PRE>
public static final int <B>YEAST_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.YEAST_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="MOLD_PROTOZOAN_MT_ID"><!-- --></A><H3>
MOLD_PROTOZOAN_MT_ID</H3>
<PRE>
public static final int <B>MOLD_PROTOZOAN_MT_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.MOLD_PROTOZOAN_MT_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="MYCOPLASMA_ID"><!-- --></A><H3>
MYCOPLASMA_ID</H3>
<PRE>
public static final int <B>MYCOPLASMA_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.MYCOPLASMA_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="INVERTEBRATE_MT_ID"><!-- --></A><H3>
INVERTEBRATE_MT_ID</H3>
<PRE>
public static final int <B>INVERTEBRATE_MT_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.INVERTEBRATE_MT_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="CILIATE_ID"><!-- --></A><H3>
CILIATE_ID</H3>
<PRE>
public static final int <B>CILIATE_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.CILIATE_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="ECHINODERM_MT_ID"><!-- --></A><H3>
ECHINODERM_MT_ID</H3>
<PRE>
public static final int <B>ECHINODERM_MT_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.ECHINODERM_MT_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="EUPLOTID_NUC_ID"><!-- --></A><H3>
EUPLOTID_NUC_ID</H3>
<PRE>
public static final int <B>EUPLOTID_NUC_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.EUPLOTID_NUC_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="BACTERIAL_ID"><!-- --></A><H3>
BACTERIAL_ID</H3>
<PRE>
public static final int <B>BACTERIAL_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.BACTERIAL_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="ALT_YEAST_ID"><!-- --></A><H3>
ALT_YEAST_ID</H3>
<PRE>
public static final int <B>ALT_YEAST_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.ALT_YEAST_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="ASCIDIAN_MT_ID"><!-- --></A><H3>
ASCIDIAN_MT_ID</H3>
<PRE>
public static final int <B>ASCIDIAN_MT_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.ASCIDIAN_MT_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="FLATWORM_MT_ID"><!-- --></A><H3>
FLATWORM_MT_ID</H3>
<PRE>
public static final int <B>FLATWORM_MT_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.FLATWORM_MT_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="BLEPHARISMA_NUC_ID"><!-- --></A><H3>
BLEPHARISMA_NUC_ID</H3>
<PRE>
public static final int <B>BLEPHARISMA_NUC_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.BLEPHARISMA_NUC_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="NO_STOPS_ID"><!-- --></A><H3>
NO_STOPS_ID</H3>
<PRE>
public static final int <B>NO_STOPS_ID</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#dr.evolution.datatype.GeneticCode.NO_STOPS_ID">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="GENETIC_CODE_TABLES"><!-- --></A><H3>
GENETIC_CODE_TABLES</H3>
<PRE>
public static final java.lang.String[] <B>GENETIC_CODE_TABLES</B></PRE>
<DL>
<DD>Standard genetic code tables from GENBANK
 Nucleotides go A, C, G, T - Note: this is not the order used by the Genbank web site
 With the first codon position most significant (i.e. AAA, AAC, AAG, AAT, ACA, etc.).
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="GENETIC_CODE_NAMES"><!-- --></A><H3>
GENETIC_CODE_NAMES</H3>
<PRE>
public static final java.lang.String[] <B>GENETIC_CODE_NAMES</B></PRE>
<DL>
<DD>Names of the standard genetic code tables from GENBANK
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="GENETIC_CODE_DESCRIPTIONS"><!-- --></A><H3>
GENETIC_CODE_DESCRIPTIONS</H3>
<PRE>
public static final java.lang.String[] <B>GENETIC_CODE_DESCRIPTIONS</B></PRE>
<DL>
<DD>Descriptions of the standard genetic code tables from GENBANK
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="UNIVERSAL"><!-- --></A><H3>
UNIVERSAL</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>UNIVERSAL</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="VERTEBRATE_MT"><!-- --></A><H3>
VERTEBRATE_MT</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>VERTEBRATE_MT</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="YEAST"><!-- --></A><H3>
YEAST</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>YEAST</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="MOLD_PROTOZOAN_MT"><!-- --></A><H3>
MOLD_PROTOZOAN_MT</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>MOLD_PROTOZOAN_MT</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="MYCOPLASMA"><!-- --></A><H3>
MYCOPLASMA</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>MYCOPLASMA</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="INVERTEBRATE_MT"><!-- --></A><H3>
INVERTEBRATE_MT</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>INVERTEBRATE_MT</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="CILIATE"><!-- --></A><H3>
CILIATE</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>CILIATE</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="ECHINODERM_MT"><!-- --></A><H3>
ECHINODERM_MT</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>ECHINODERM_MT</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="EUPLOTID_NUC"><!-- --></A><H3>
EUPLOTID_NUC</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>EUPLOTID_NUC</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="BACTERIAL"><!-- --></A><H3>
BACTERIAL</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>BACTERIAL</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="ALT_YEAST"><!-- --></A><H3>
ALT_YEAST</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>ALT_YEAST</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="ASCIDIAN_MT"><!-- --></A><H3>
ASCIDIAN_MT</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>ASCIDIAN_MT</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="FLATWORM_MT"><!-- --></A><H3>
FLATWORM_MT</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>FLATWORM_MT</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="BLEPHARISMA_NUC"><!-- --></A><H3>
BLEPHARISMA_NUC</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>BLEPHARISMA_NUC</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="NO_STOPS"><!-- --></A><H3>
NO_STOPS</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A> <B>NO_STOPS</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="GENETIC_CODES"><!-- --></A><H3>
GENETIC_CODES</H3>
<PRE>
public static final <A HREF="../../../dr/evolution/datatype/GeneticCode.html" title="class in dr.evolution.datatype">GeneticCode</A>[] <B>GENETIC_CODES</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="geneticCode"><!-- --></A><H3>
geneticCode</H3>
<PRE>
private int <B>geneticCode</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="codeTable"><!-- --></A><H3>
codeTable</H3>
<PRE>
private java.lang.String <B>codeTable</B></PRE>
<DL>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="GeneticCode(int)"><!-- --></A><H3>
GeneticCode</H3>
<PRE>
public <B>GeneticCode</B>(int&nbsp;geneticCode)</PRE>
<DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getName()"><!-- --></A><H3>
getName</H3>
<PRE>
public java.lang.String <B>getName</B>()</PRE>
<DL>
<DD>Returns the name of the genetic code
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getDescription()"><!-- --></A><H3>
getDescription</H3>
<PRE>
public java.lang.String <B>getDescription</B>()</PRE>
<DL>
<DD>Returns the description of the genetic code
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getAminoAcidChar(int)"><!-- --></A><H3>
getAminoAcidChar</H3>
<PRE>
public char <B>getAminoAcidChar</B>(int&nbsp;codonState)</PRE>
<DL>
<DD>Returns the char associated with AminoAcid represented by codonState.
 Note that the char is as defined by AminoAcids.java
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html#getAminoAcidChar(int)">getAminoAcidChar</A></CODE> in interface <CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>state for '?' if codon unknown<DT><B>See Also:</B><DD><A HREF="../../../dr/evolution/datatype/AminoAcids.html" title="class in dr.evolution.datatype"><CODE>AminoAcids</CODE></A>, 
<A HREF="../../../dr/evolution/datatype/Codons.html" title="class in dr.evolution.datatype"><CODE>Codons</CODE></A></DL>
</DD>
</DL>
<HR>

<A NAME="getAminoAcidState(int)"><!-- --></A><H3>
getAminoAcidState</H3>
<PRE>
public int <B>getAminoAcidState</B>(int&nbsp;codonState)</PRE>
<DL>
<DD>Returns the state associated with AminoAcid represented by codonState.
 Note that the state is the canonical state (generated combinatorially)
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html#getAminoAcidState(int)">getAminoAcidState</A></CODE> in interface <CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>'?' if codon unknown<DT><B>See Also:</B><DD><A HREF="../../../dr/evolution/datatype/AminoAcids.html" title="class in dr.evolution.datatype"><CODE>AminoAcids</CODE></A>, 
<A HREF="../../../dr/evolution/datatype/Codons.html" title="class in dr.evolution.datatype"><CODE>Codons</CODE></A></DL>
</DD>
</DL>
<HR>

<A NAME="isStopCodon(int)"><!-- --></A><H3>
isStopCodon</H3>
<PRE>
public boolean <B>isStopCodon</B>(int&nbsp;codonState)</PRE>
<DL>
<DD>Note that the state is the canonical state (generated combinatorially)
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>whether the codonState is a stop codon</DL>
</DD>
</DL>
<HR>

<A NAME="getCodonsFromAminoAcidState(int)"><!-- --></A><H3>
getCodonsFromAminoAcidState</H3>
<PRE>
public char[][] <B>getCodonsFromAminoAcidState</B>(int&nbsp;aminoAcidState)</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html#getCodonsFromAminoAcidState(int)">getCodonsFromAminoAcidState</A></CODE> in interface <CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>all the possible codons for a given amino acid</DL>
</DD>
</DL>
<HR>

<A NAME="getCodonsFromAminoAcidChar(char)"><!-- --></A><H3>
getCodonsFromAminoAcidChar</H3>
<PRE>
public char[][] <B>getCodonsFromAminoAcidChar</B>(char&nbsp;aminoAcidChar)</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html#getCodonsFromAminoAcidChar(char)">getCodonsFromAminoAcidChar</A></CODE> in interface <CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getAmbiguousCodonFromAminoAcidState(int)"><!-- --></A><H3>
getAmbiguousCodonFromAminoAcidState</H3>
<PRE>
public int[] <B>getAmbiguousCodonFromAminoAcidState</B>(int&nbsp;aminoAcid)</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html#getAmbiguousCodonFromAminoAcidState(int)">getAmbiguousCodonFromAminoAcidState</A></CODE> in interface <CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getStopCodonIndices()"><!-- --></A><H3>
getStopCodonIndices</H3>
<PRE>
public int[] <B>getStopCodonIndices</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html#getStopCodonIndices()">getStopCodonIndices</A></CODE> in interface <CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the codon states of stop amino acids.</DL>
</DD>
</DL>
<HR>

<A NAME="getStopCodonCount()"><!-- --></A><H3>
getStopCodonCount</H3>
<PRE>
public int <B>getStopCodonCount</B>()</PRE>
<DL>
<DD>Returns the number of terminator amino acids.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html#getStopCodonCount()">getStopCodonCount</A></CODE> in interface <CODE><A HREF="../../../dr/evolution/datatype/CodonTable.html" title="interface in dr.evolution.datatype">CodonTable</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/GeneticCode.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../dr/evolution/datatype/GeneralDataType.State.html" title="class in dr.evolution.datatype"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../dr/evolution/datatype/HiddenCodons.html" title="class in dr.evolution.datatype"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?dr/evolution/datatype/GeneticCode.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="GeneticCode.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
