<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_45) on Fri Jun 07 16:03:57 EDT 2013 -->
<TITLE>
Uses of Package dr.evolution.datatype
</TITLE>

<META NAME="date" CONTENT="2013-06-07">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Package dr.evolution.datatype";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?dr/evolution/datatype/package-use.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="package-use.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Package<br>dr.evolution.datatype</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.beagle.evomodel.parsers"><B>dr.app.beagle.evomodel.parsers</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.beagle.evomodel.substmodel"><B>dr.app.beagle.evomodel.substmodel</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.beagle.evomodel.treelikelihood"><B>dr.app.beagle.evomodel.treelikelihood</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.beagle.tools"><B>dr.app.beagle.tools</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.beauti.generator"><B>dr.app.beauti.generator</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.beauti.options"><B>dr.app.beauti.options</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.beauti.siteModelsPanel"><B>dr.app.beauti.siteModelsPanel</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.misc"><B>dr.app.misc</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.seqgen"><B>dr.app.seqgen</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.app.tools"><B>dr.app.tools</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evolution.alignment"><B>dr.evolution.alignment</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evolution.datatype"><B>dr.evolution.datatype</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evolution.distance"><B>dr.evolution.distance</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evolution.io"><B>dr.evolution.io</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evolution.sequence"><B>dr.evolution.sequence</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evolution.tree"><B>dr.evolution.tree</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evolution.wrightfisher"><B>dr.evolution.wrightfisher</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evomodel.MSSD"><B>dr.evomodel.MSSD</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evomodel.arg.likelihood"><B>dr.evomodel.arg.likelihood</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evomodel.branchratemodel"><B>dr.evomodel.branchratemodel</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evomodel.substmodel"><B>dr.evomodel.substmodel</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evomodel.tree"><B>dr.evomodel.tree</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evomodel.treelikelihood"><B>dr.evomodel.treelikelihood</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.evoxml.util"><B>dr.evoxml.util</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#dr.inference.markovjumps"><B>dr.inference.markovjumps</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#test.dr.evomodel.substmodel"><B>test.dr.evomodel.substmodel</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#test.dr.evomodel.treelikelihood"><B>test.dr.evomodel.treelikelihood</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#test.dr.inference.trace"><B>test.dr.inference.trace</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.beagle.evomodel.parsers"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/beagle/evomodel/parsers/package-summary.html">dr.app.beagle.evomodel.parsers</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.app.beagle.evomodel.parsers"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.beagle.evomodel.substmodel"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/beagle/evomodel/substmodel/package-summary.html">dr.app.beagle.evomodel.substmodel</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Codons.html#dr.app.beagle.evomodel.substmodel"><B>Codons</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements DataType for codons
 <p/>
 Codons have tree different representations:
 State numbers - 0-63 + 64, 65 as unknown and gap
 State chars - the above converted into chars, starting at 'A'
 and '?' + '-' for unknown and gap
 Strings or chars of three nucleotide characters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.app.beagle.evomodel.substmodel"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/GeneticCode.html#dr.app.beagle.evomodel.substmodel"><B>GeneticCode</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A set of standard genetic codes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/HiddenCodons.html#dr.app.beagle.evomodel.substmodel"><B>HiddenCodons</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.beagle.evomodel.treelikelihood"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/beagle/evomodel/treelikelihood/package-summary.html">dr.app.beagle.evomodel.treelikelihood</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.app.beagle.evomodel.treelikelihood"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.beagle.tools"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/beagle/tools/package-summary.html">dr.app.beagle.tools</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.app.beagle.tools"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.beauti.generator"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/beauti/generator/package-summary.html">dr.app.beauti.generator</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Microsatellite.html#dr.app.beauti.generator"><B>Microsatellite</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.beauti.options"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/beauti/options/package-summary.html">dr.app.beauti.options</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.app.beauti.options"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Microsatellite.html#dr.app.beauti.options"><B>Microsatellite</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/PloidyType.html#dr.app.beauti.options"><B>PloidyType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.beauti.siteModelsPanel"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/beauti/siteModelsPanel/package-summary.html">dr.app.beauti.siteModelsPanel</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.app.beauti.siteModelsPanel"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.misc"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/misc/package-summary.html">dr.app.misc</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.app.misc"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.seqgen"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/seqgen/package-summary.html">dr.app.seqgen</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Microsatellite.html#dr.app.seqgen"><B>Microsatellite</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.app.tools"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/app/tools/package-summary.html">dr.app.tools</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.app.tools"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evolution.alignment"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evolution/alignment/package-summary.html">dr.evolution.alignment</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/CodonTable.html#dr.evolution.alignment"><B>CodonTable</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Describes a device for translating Nucleotide triplets
 or codon indices into amino acid codes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evolution.alignment"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/PairedDataType.html#dr.evolution.alignment"><B>PairedDataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;implements DataType for nucleotides with hidden state(s)</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evolution.datatype"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/AminoAcids.html#dr.evolution.datatype"><B>AminoAcids</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;implements DataType for amino acids.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Codons.html#dr.evolution.datatype"><B>Codons</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements DataType for codons
 <p/>
 Codons have tree different representations:
 State numbers - 0-63 + 64, 65 as unknown and gap
 State chars - the above converted into chars, starting at 'A'
 and '?' + '-' for unknown and gap
 Strings or chars of three nucleotide characters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/CodonTable.html#dr.evolution.datatype"><B>CodonTable</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Describes a device for translating Nucleotide triplets
 or codon indices into amino acid codes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/ContinuousDataType.html#dr.evolution.datatype"><B>ContinuousDataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evolution.datatype"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/GeneralDataType.html#dr.evolution.datatype"><B>GeneralDataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements a general DataType for any number of states</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/GeneralDataType.State.html#dr.evolution.datatype"><B>GeneralDataType.State</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/GeneticCode.html#dr.evolution.datatype"><B>GeneticCode</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A set of standard genetic codes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/HiddenCodons.html#dr.evolution.datatype"><B>HiddenCodons</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/HiddenDataType.html#dr.evolution.datatype"><B>HiddenDataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/HiddenNucleotides.html#dr.evolution.datatype"><B>HiddenNucleotides</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;implements DataType for nucleotides with hidden state(s)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Microsatellite.html#dr.evolution.datatype"><B>Microsatellite</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/NewHiddenNucleotides.html#dr.evolution.datatype"><B>NewHiddenNucleotides</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Nucleotides.html#dr.evolution.datatype"><B>Nucleotides</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;implements DataType for nucleotides with ambiguous characters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/P2P.html#dr.evolution.datatype"><B>P2P</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/PairedDataType.html#dr.evolution.datatype"><B>PairedDataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;implements DataType for nucleotides with hidden state(s)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/PloidyType.html#dr.evolution.datatype"><B>PloidyType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/SimpleCodons.html#dr.evolution.datatype"><B>SimpleCodons</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements DataType for codons
 
 Codons have tree different representations: 
        State numbers - 0-63 + 64, 65 as unknown and gap
  State chars - the above converted into chars, starting at 'A'
                and '?' + '-' for unknown and gap
        Strings or chars of three nucleotide characters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/TwoStateCovarion.html#dr.evolution.datatype"><B>TwoStateCovarion</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/TwoStates.html#dr.evolution.datatype"><B>TwoStates</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;implements DataType for two-state data</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evolution.distance"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evolution/distance/package-summary.html">dr.evolution.distance</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evolution.distance"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evolution.io"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evolution/io/package-summary.html">dr.evolution.io</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evolution.io"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Microsatellite.html#dr.evolution.io"><B>Microsatellite</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evolution.sequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evolution/sequence/package-summary.html">dr.evolution.sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evolution.sequence"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evolution.tree"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evolution/tree/package-summary.html">dr.evolution.tree</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evolution.tree"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evolution.wrightfisher"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evolution/wrightfisher/package-summary.html">dr.evolution.wrightfisher</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Codons.html#dr.evolution.wrightfisher"><B>Codons</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements DataType for codons
 <p/>
 Codons have tree different representations:
 State numbers - 0-63 + 64, 65 as unknown and gap
 State chars - the above converted into chars, starting at 'A'
 and '?' + '-' for unknown and gap
 Strings or chars of three nucleotide characters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/GeneticCode.html#dr.evolution.wrightfisher"><B>GeneticCode</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A set of standard genetic codes.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evomodel.MSSD"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evomodel/MSSD/package-summary.html">dr.evomodel.MSSD</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/MutationDeathType.html#dr.evomodel.MSSD"><B>MutationDeathType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Package: MutationDeathType
 Description:
 <p/>
 <p/>
 Created by
 Alexander V.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evomodel.arg.likelihood"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evomodel/arg/likelihood/package-summary.html">dr.evomodel.arg.likelihood</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evomodel.arg.likelihood"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evomodel.branchratemodel"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evomodel/branchratemodel/package-summary.html">dr.evomodel.branchratemodel</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evomodel.branchratemodel"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evomodel.substmodel"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evomodel/substmodel/package-summary.html">dr.evomodel.substmodel</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Codons.html#dr.evomodel.substmodel"><B>Codons</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements DataType for codons
 <p/>
 Codons have tree different representations:
 State numbers - 0-63 + 64, 65 as unknown and gap
 State chars - the above converted into chars, starting at 'A'
 and '?' + '-' for unknown and gap
 Strings or chars of three nucleotide characters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evomodel.substmodel"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/GeneticCode.html#dr.evomodel.substmodel"><B>GeneticCode</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A set of standard genetic codes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/HiddenNucleotides.html#dr.evomodel.substmodel"><B>HiddenNucleotides</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;implements DataType for nucleotides with hidden state(s)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Microsatellite.html#dr.evomodel.substmodel"><B>Microsatellite</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/TwoStateCovarion.html#dr.evomodel.substmodel"><B>TwoStateCovarion</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evomodel.tree"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evomodel/tree/package-summary.html">dr.evomodel.tree</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evomodel.tree"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evomodel.treelikelihood"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evomodel/treelikelihood/package-summary.html">dr.evomodel.treelikelihood</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evomodel.treelikelihood"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.evoxml.util"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/evoxml/util/package-summary.html">dr.evoxml.util</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.evoxml.util"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="dr.inference.markovjumps"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../dr/inference/markovjumps/package-summary.html">dr.inference.markovjumps</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#dr.inference.markovjumps"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="test.dr.evomodel.substmodel"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../test/dr/evomodel/substmodel/package-summary.html">test.dr.evomodel.substmodel</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Codons.html#test.dr.evomodel.substmodel"><B>Codons</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements DataType for codons
 <p/>
 Codons have tree different representations:
 State numbers - 0-63 + 64, 65 as unknown and gap
 State chars - the above converted into chars, starting at 'A'
 and '?' + '-' for unknown and gap
 Strings or chars of three nucleotide characters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#test.dr.evomodel.substmodel"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/GeneralDataType.html#test.dr.evomodel.substmodel"><B>GeneralDataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements a general DataType for any number of states</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/GeneticCode.html#test.dr.evomodel.substmodel"><B>GeneticCode</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A set of standard genetic codes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/HiddenNucleotides.html#test.dr.evomodel.substmodel"><B>HiddenNucleotides</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;implements DataType for nucleotides with hidden state(s)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/Microsatellite.html#test.dr.evomodel.substmodel"><B>Microsatellite</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="test.dr.evomodel.treelikelihood"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../test/dr/evomodel/treelikelihood/package-summary.html">test.dr.evomodel.treelikelihood</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#test.dr.evomodel.treelikelihood"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="test.dr.inference.trace"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../dr/evolution/datatype/package-summary.html">dr.evolution.datatype</A> used by <A HREF="../../../test/dr/inference/trace/package-summary.html">test.dr.inference.trace</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../dr/evolution/datatype/class-use/DataType.html#test.dr.inference.trace"><B>DataType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Base class for sequence data types.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?dr/evolution/datatype/package-use.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="package-use.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
